Core/MemMapHelpers.h

```diff
@@ -28,31 +30,34 @@ extern MIPSState *currentMIPS;
 namespace Memory
 {
 
-inline void Memcpy(const u32 to_address, const void *from_data, const u32 len, const std::string &tag = "Memcpy") {
+inline void Memcpy(const u32 to_address, const void *from_data, const u32 len, const char *tag = "Memcpy") {
 	u8 *to = GetPointer(to_address);
 	if (to) {
 		memcpy(to, from_data, len);
-		NotifyMemInfo(MemBlockFlags::WRITE, to_address, len, tag);
+		size_t tagLen = strlen(tag);
```

Comment:
Really a shame if std::string is not optimizing - I would really expect it to be in the internal string buffer, although I guess it's always best to assume STL is terrible somewhere.  Are we sure the tag names are not just too long and getting truncated now?

-[Unknown]

Reply:
Empirically It seems to be doing a horrible job, and even more so in debug mode.

The internal string buffer is a lot smaller than most of our strings. It's just a few bytes, right? And you can't even trust it to be there, really...

Reply:
https://shaharmike.com/cpp/std-string/

MSVC: 16 bytes
GCC: 16
clang: 22

I know that this isn't guaranteed, but neither is `memcpy()` optimizing to a direct read instead of a call.  It's all compiler optimizations all the way down, really.

-[Unknown]

Reply:
Hm, bigger than I thought. But profiles don't lie...